pkg/query-service/app/http_handler.go

```diff
@@ -1988,6 +1992,92 @@ func (aH *APIHandler) getCustomRetentionTTL(w http.ResponseWriter, r *http.Reque
 	aH.WriteJSON(w, r, result)
 }
 
+// promotePaths inserts provided JSON paths into the promoted paths table in ClickHouse.
+func (aH *APIHandler) promotePaths(w http.ResponseWriter, r *http.Request) {
+	ctx := r.Context()
+	claims, errv2 := authtypes.ClaimsFromContext(ctx)
+	if errv2 != nil {
+		render.Error(w, errorsV2.Newf(errorsV2.TypeInternal, errorsV2.CodeInternal, "failed to get org id from context"))
+		return
+	}
+
+	var resp any
+	var err error
+	switch r.Method {
+	case http.MethodGet:
+		err := func() error {
+			indexes, err := aH.reader.ListBodySkipIndexes(ctx)
+			if err != nil {
+				return err
+			}
+			aggr := map[string][]schemamigrator.Index{}
+			for _, index := range indexes {
+				path, err := schemamigrator.UnfoldJSONSubColumnIndexExpr(index.Expression)
+				if err != nil {
+					return err
+				}
+				aggr[path] = append(aggr[path], index)
+			}
+			promotedPaths, err := aH.reader.ListPromotedPaths(ctx)
+			if err != nil {
+				return err
+			}
+
+			response := []model.PromotePathItem{}
+			for _, path := range promotedPaths {
+				fullPath := telemetrylogs.BodyPromotedColumnPrefix + path
+				path = telemetrylogs.BodyJSONStringSearchPrefix + path
+				item := model.PromotePathItem{
+					Path:    path,
+					Promote: true,
+				}
+				indexes, ok := aggr[fullPath]
+				if ok {
+					item.Indexes = indexes
+					delete(aggr, fullPath)
+				}
+				response = append(response, item)
+			}
+
+			// add the paths that are not promoted but have indexes
+			for _, indexes := range aggr {
+				expr := indexes[0].Expression
+				expr, err := schemamigrator.UnfoldJSONSubColumnIndexExpr(expr)
+				if err != nil {
+					return err
+				}
+				path := strings.TrimPrefix(expr, telemetrylogs.BodyJSONColumnPrefix)
+				path = telemetrylogs.BodyJSONStringSearchPrefix + path
+				response = append(response, model.PromotePathItem{
+					Path:    path,
+					Indexes: indexes,
+				})
+			}
+			resp = response
+			return nil
+		}()
+		if err != nil {
```

Comment:
can you share some example response with all two cases

that is promoted path with index and path